src/couch_changes.erl

```diff
@@ -336,6 +343,16 @@ get_doc_ids(_) ->
     throw({bad_request, no_doc_ids_provided}).
 
 
+get_selector({json_req, {Props}}) ->
+    check_selector(couch_util:get_value(<<"selector">>, Props));
+get_selector(#httpd{method='POST'}=Req) ->
+    couch_httpd:validate_ctype(Req, "application/json"),
+    {Props} = couch_httpd:json_body_obj(Req),
+    check_selector(couch_util:get_value(<<"selector">>, Props));
+get_selector(_) ->
+    throw({bad_request, invalid_selector}).
```

Comment:
It is an invalid_selector because this check happens after noticing there is a _selector as filter. Which prompted the the call to `configure_filter("_selector",...`. At this point if it is not  GET request, or object not is an array it is an "invalid selector".